packages/firestore/test/unit/local/web_storage_shared_client_state.test.ts

```diff
@@ -60,49 +65,97 @@ function mutationKey(user: User, batchId: BatchId): string {
   }
 }
 
+function targetKey(targetId: TargetId): string {
+  return `fs_targets_${persistenceHelpers.TEST_PERSISTENCE_PREFIX}_${targetId}`;
+}
+
 interface TestSharedClientState {
   pendingBatches: BatchId[];
   acknowledgedBatches: BatchId[];
   rejectedBatches: { [batchId: number]: FirestoreError };
+  pendingTargets: TargetId[];
+  activeTargets: TargetId[];
+  currentTargets: TargetId[];
+  rejectedTargets: { [batchId: number]: FirestoreError };
 }
 
 /**
  * Implementation of `SharedClientStateSyncer` that aggregates its callback
  * data and exposes it via `.sharedClientState`.
  */
 class TestSharedClientSyncer implements SharedClientStateSyncer {
-  private pendingBatches: BatchId[] = [];
-  private acknowledgedBatches: BatchId[] = [];
-  private rejectedBatches: { [batchId: number]: FirestoreError } = {};
+  private mutationBatches: {
+    [batchId: number]: { state: MutationBatchState; error?: FirestoreError };
+  } = {};
+  private queryTargets: {
+    [targetId: number]: { state: QueryTargetState; error?: FirestoreError };
+  } = {};
 
   constructor(public activeClients: ClientId[]) {}
 
   get sharedClientState(): TestSharedClientState {
-    return {
-      pendingBatches: this.pendingBatches,
-      acknowledgedBatches: this.acknowledgedBatches,
-      rejectedBatches: this.rejectedBatches
+    const state: TestSharedClientState = {
+      pendingBatches: [],
+      acknowledgedBatches: [],
+      rejectedBatches: {},
+      pendingTargets: [],
+      activeTargets: [],
+      currentTargets: [],
+      rejectedTargets: {}
     };
```

Comment:
I don't care too strongly but it seems like this is optimizing for having a bunch of targets and then being able to assert that they end up in the right states like `expect(clientState.activeTargets).to.equal([1, 5, 7, 8])` but I think all of your tests only deal with a single target and you end up doing:

```
        expect(clientState.pendingTargets).to.be.empty;
        expect(clientState.activeTargets).to.be.empty;
        expect(clientState.currentTargets).to.have.members([1]);
        expect(clientState.rejectedTargets).to.be.empty;
```

And so it'd be simpler if you could just do:

```
expect(clientState.targetStates[1]).to.equal('pending')
```

Again I don't care that much, but this seems roundabout and more complicated than necessary given your current tests.

Reply:
I went ahead with this and updated the test assertions to directly assert the states on the map that I built up during the test run. These tests only exercise a single step at a time, so most of the assertions are fairly simple. The previous tests did a better job of hiding this :)